### ISSUE TITLE
rubocop --config with "Exclude" incorrectly excludes config folder path instead of working folder

### ISSUE THREAD

**Issue by bbugh** [bbatsov/rubocop#4621]
When rubocop is run from a directory and specifies the `--config` file in another directory, "Exclude" is factored from the *config* folder, and not the *working* folder. 

Given a `.rubocop.yml` file:

```yaml
# ~/some/project/folder/.rubocop.yml
Style/BlockDelimiters:
  Exclude:
    - "spec/**/*"
```

When changing the current working directory to a temp folder:

```
cd /var/folders/sb/_lcgg2pd0r10fynq3wd206v00000gn/T/rubocop-beautify117620-5114-1fq86oj.hdjxl84cxr
```

And running rubocop from the temp directory:

```
rubocop-beautify117620-5114-1fq86oj.hdjxl84cxr » rubocop --force-exclusion 
  --config ~/some/project/folder/.rubocop.yml
  spec/request/some_controller_spec.rb                         # should be excluded
  ~/some/project/folder/spec/request/some_controller_spec.rb   # should not be excluded
```

The result shows that the *working* directory is not excluded, while the *configuration file* directory is excluded. This is backwards!

```
Inspecting 2 files
CC

Offenses:

spec/request/some_controller_spec.rb:3:1: C: Block has too many lines. [57/25]
describe SomeController, type: :request do ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
spec/request/some_controller_spec.rb:9:3: C: Block has too many lines. [51/25]
  context "with a logged in user" do ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
spec/request/some_controller_spec.rb:43:18: C: Avoid using {...} for multi-line blocks.
          expect {
                 ^
/Users/bbugh/project_folder/spec/request/some_controller_spec.rb:3:1: C: Block has too many lines. [57/25]
describe SomeController, type: :request do ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/bbugh/project_folder/spec/request/some_controller_spec.rb:9:3: C: Block has too many lines. [51/25]
  context "with a logged in user" do ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

2 files inspected, 5 offenses detected
```

I encountered this issue while fixing [atom-beautify/issues/1773](https://github.com/Glavin001/atom-beautify/issues/1773). 

--------

## Expected behavior

When passing in the `--config` switch, I expect Rubocop to exclude patterns based on the working directory where it is run, not based on the path of the config file.

(I understand this may be partially intentional so rubocop can be run from any subdirectory of the main project, but I believe this behavior should change when manually specifying the config.)

## Actual behavior

The files that are expected to be excluded by the specified cop are not, and vice versa.

## Steps to reproduce the problem

1. Add this to the project `.rubocop.yml`
    ```ruby
    Style/BlockDelimiters:
      Exclude:
        - "spec/**/*"
    ```

1. Create a file like this under `project/spec`:
    ```ruby
    # /spec/example_spec.rb
    describe "something" do
      it "raises an error" do 
        expect {
          # whatever
        }.to raise_error ArgumentError
      end
    end
    ```

2. Copy the spec file to a tmp folder `/tmp/spec/example_spec.rb`
3. `cd /tmp/`
4. `rubocop --config ~/project/.rubocop.yml spec/example_spec.rb ~/project/spec/example_spec.rb`

The incorrect Exclude behavior will be demonstrated.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.3.0 x86_64-darwin14)
```

**Comment by jonas054** [comment]
The behavior is intended, and is described in http://rubocop.readthedocs.io/en/latest/configuration/#includingexcluding-files

> In .rubocop.yml and any other configuration file beginning with .rubocop, files and directories are specified relative to the directory where the configuration file is. In configuration files that don't begin with .rubocop, e.g. our_company_defaults.yml, paths are relative to the directory where rubocop is run.

I'm worried that complexity will increase and possibly exceed acceptable levels if we start making exceptions to the above rule. Remember that there's also inheritance of configuration. What should the rule be when combining `--config` with inheritance?

You can work around the problem by giving the file a name that doesn't begin with `.rubocop`.
